src/Home/JoinRoomButton.js

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import { supabase } from "../supabase_client";
+
+const JoinRoomButton = (props) => {
+  const { player, setPage, setRoomCode } = props;
+  const [userRoomInput, setUserRoomInput] = useState();
+  const [roomExists, setRoomExists] = useState(true);
+  const onChange = (e) => {
+    setUserRoomInput(e.target.value);
+  };
+  const handleClick = async () => {
```

Comment:
```suggestion
  const [roomExists, setRoomExists] = useState(true);
  
  const onChange = (e) => {
    setUserRoomInput(e.target.value);
  };
  
  const handleClick = async () => {
```

spacing between functions to read easier